File: 02_Estructuras de Control/02_Condicionales/03_Anidadas/CSharp/ejemplo1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Ejemplo1Anidado
{
    internal class Program
    {
        static void Main(string[] args)
        {
            /* Programa para Determinar el Nivel de Capacitación de un Postulante con Indicadores Visuales por Rango de Desempeño
             */
            
            Console.ForegroundColor = ConsoleColor.DarkCyan;
            Console.WriteLine("===== CLASIFICACIÓN DEL NIVEL DE CAPACITACIÓN =====");
            Console.ResetColor();
            Console.WriteLine();

            // Entrada datos

            Console.WriteLine("Ingrese la cantidad total de preguntas realizadas: ");
            int totalpreguntas = int.Parse(Console.ReadLine());

            Console.WriteLine();
            Console.WriteLine("Ingrese la cantidad de preguntas contestadas correctamente: ");
            int correctas = int.Parse(Console.ReadLine());

            // Proceso

            if (totalpreguntas <= 0 || correctas < 0 || correctas > totalpreguntas)
            {
                Console.WriteLine("Error: Datos inválidos. Asegúrese de ingresar valores correctos. ");
                return;
            }

            double porcentaje = (double)correctas / totalpreguntas * 100;

            string nivel;

            if (porcentaje >= 90.0)
            {
                nivel = "Nivel máximo";
                Console.ForegroundColor = ConsoleColor.Green;
            }

            else if (porcentaje >= 75.0)
            {
                nivel = "Nivle medio";
                Console.ForegroundColor = ConsoleColor.Cyan;
            }

            else if (porcentaje >= 50.0)
            {
                nivel = "Nivel regular";
                Console.ForegroundColor = ConsoleColor.Yellow;
            }

            else
            {
                nivel = "Fuera de nivel";
                Console.ForegroundColor = ConsoleColor.Red;
            }

            Console.WriteLine();
            Console.WriteLine($"Resultados:");
            Console.WriteLine($"Total de preguntas: {totalpreguntas}");
            Console.WriteLine($"Respuestas correctas: {correctas}");
            Console.WriteLine($"Porcentaje obtenido: {porcentaje:F2}%");
            Console.WriteLine($"Nivel asignado: {nivel}");

            Console.ResetColor();

            Console.WriteLine("\nPresione cualquier tecla para salir...");
            Console.ReadKey();
        }
    }
}

File: 02_Estructuras de Control/02_Condicionales/03_Anidadas/Python/ejemplo1.py
# Programa para Determinar el Nivel de Capacitación de un Postulante según su Porcentaje de Respuestas Correctas

print("===== CLASIFICACIÓN DEL NIVEL DE CAPACITACIÓN =====")
print()

# Entrada de datos
try:
    total_preguntas = int(input("Ingrese la cantidad total de preguntas realizadas: "))
    correctas = int(input("Ingrese la cantidad de preguntas contestadas correctamente: "))
except ValueError:
    print("Error: Debe ingresar números enteros.")
    exit()

# Validación de datos
if total_preguntas <= 0 or correctas < 0 or correctas > total_preguntas:
    print("Error: Datos inválidos. Asegúrese de ingresar valores correctos.")
    exit()

# Cálculo del porcentaje
porcentaje = (correctas / total_preguntas) * 100

# Determinación del nivel (estructura condicional anidada en cascada)
if porcentaje >= 90.0:
    nivel = "Nivel máximo"
elif porcentaje >= 75.0:
    nivel = "Nivel medio"
elif porcentaje >= 50.0:
    nivel = "Nivel regular"
else:
    nivel = "Fuera de nivel"

# Salida de resultados
print()
print("Resultados:")
print(f"Total de preguntas: {total_preguntas}")
print(f"Respuestas correctas: {correctas}")
print(f"Porcentaje obtenido: {porcentaje:.2f}%")
print(f"Nivel asignado: {nivel}")